src/biogeophys/SnowSnicarMod.F90

```diff
@@ -578,81 +780,235 @@ subroutine SNICAR_RT (flg_snw_ice, bounds, num_nourbanc, filter_nourbanc,  &
+                           AR_tmp = sno_AR(i)  ! user specified value
+                        endif
+                        do igb = 1,7
+                           g_ice_Cg_tmp(igb) = g_b0(igb) * ((fs_hex0/fs_hex)**g_b1(igb)) * (diam_ice**g_b2(igb))   ! Eq.7, He et al. (2017)
+                           gg_ice_F07_tmp(igb) = g_F07_p0(igb) + g_F07_p1(igb) * log(AR_tmp) + g_F07_p2(igb) * (log(AR_tmp) * log(AR_tmp)) ! Eqn. 3.3 in Fu (2007)
+                        enddo
+
+                     case ('koch_snowflake')
+                        diam_ice = 2._r8 * snw_rds_lcl(i) / 0.544_r8  ! unit: microns
+                        if (sno_fs(i) == 0._r8) then
+                           fs_koch = 0.712_r8  ! default; He et al. (2017), Table 1
+                        else
+                           fs_koch = sno_fs(i) ! user specified value
+                        endif
+                        fs_hex = 0.788_r8      ! reference shape factor
+                        if (sno_AR(i) == 0._r8) then
+                           AR_tmp = 2.5_r8     ! default; He et al. (2017), Table 1
+                        else
+                           AR_tmp = sno_AR(i)  ! user specified value
+                        endif
+                        do igb = 1,7
+                           g_ice_Cg_tmp(igb) = g_b0(igb) * ((fs_koch/fs_hex)**g_b1(igb)) * (diam_ice**g_b2(igb))   ! Eq.7, He et al. (2017)
+                           gg_ice_F07_tmp(igb) = g_F07_p0(igb) + g_F07_p1(igb) * log(AR_tmp) + g_F07_p2(igb) * (log(AR_tmp) * log(AR_tmp)) ! Eqn. 3.3 in Fu (2007)
+                        enddo
+
+                     end select
+
+                     ! compute nonspherical snow asymmetry factor
+                     if (sno_shp(i) /= 'sphere') then
+                        ! 7 wavelength bands for g_ice to be interpolated into targeted SNICAR bands here
+                        ! use the piecewise linear interpolation subroutine created at the end of this module
+                        ! tests showed the piecewise linear interpolation has similar results as pchip interpolation
+                        call piecewise_linear_interp1d(7, g_wvl_ct, g_ice_Cg_tmp, wvl_ct(bnd_idx), g_Cg_intp)
+                        call piecewise_linear_interp1d(7, g_wvl_ct, gg_ice_F07_tmp, wvl_ct(bnd_idx), gg_F07_intp)
+                        g_ice_F07 = gg_F07_intp + 0.5_r8 * (1._r8 - gg_F07_intp) / ss_alb_snw_lcl(i)  ! Eq.2.2 in Fu (2007)
+                        asm_prm_snw_lcl(i) = g_ice_F07 * g_Cg_intp     ! Eq.6, He et al. (2017)
+                     endif
+
+                     if (asm_prm_snw_lcl(i) > 0.99_r8) asm_prm_snw_lcl(i) = 0.99_r8 !avoid unreasonable values (rarely occur in large-size spheroid cases)
+
+                  enddo ! snow layer loop
 
-                  ! aerosol species 2 optical properties
+                  ! aerosol species 2 optical properties, hydrophobic BC
                   ss_alb_aer_lcl(2)        = ss_alb_bc2(bnd_idx)      
```

Comment:
@ekluzek 
I see your point, but I'm opting against turning these indices into parameters because this is the only place where we use this particular set (aerosol species 1 through 8). Also the comments are very clear.